frontend/app/routes/public/apply/$id/adult-child/verify-email.tsx

```diff
@@ -0,0 +1,313 @@
+      });
+    }
+
+    if (state.editMode) {
+      return redirect(getPathById('public/apply/$id/adult-child/review-adult-information', params));
+    }
+
+    return redirect(getPathById('public/apply/$id/adult-child/dental-insurance', params));
+  }
+}
+
+export default function ApplyFlowVerifyEmail({ loaderData, params }: Route.ComponentProps) {
+  const { t } = useTranslation(handle.i18nNamespaces);
+  const { email, editMode } = loaderData;
+  const [showDialog, setShowDialog] = useState(false);
+
+  const fetcher = useFetcher<typeof action>();
+  const isSubmitting = fetcher.state !== 'idle';
+
+  const fetcherStatus = typeof fetcher.data === 'object' && 'status' in fetcher.data ? fetcher.data.status : undefined;
+  const errors = typeof fetcher.data === 'object' && 'errors' in fetcher.data ? fetcher.data.errors : undefined;
+  const errorSummary = useErrorSummary(errors, { verificationCode: 'verification-code' });
+
+  const communicationLink = <InlineLink routeId="public/apply/$id/adult-child/communication-preference" params={params} />;
+
+  useEffect(() => {
+    if (fetcherStatus === 'verification-code-sent') {
+      setShowDialog(true);
+    }
+  }, [fetcherStatus, fetcher.data]);
+
+  return (
+    <>
+      <div className="my-6 sm:my-8">
+        <Progress value={76} size="lg" label={t('apply:progress.label')} />
+      </div>
+      <div className="max-w-prose">
+        {fetcherStatus === 'verification-code-mismatch' && <VerificationCodeAlert />}
+        <errorSummary.ErrorSummary />
+        <fetcher.Form method="post" noValidate>
+          <CsrfTokenInput />
+          <fieldset className="mb-6">
+            <p className="mb-4">{t('apply-adult-child:verify-email.verification-code', { email })}</p>
+            <p className="mb-4">{t('apply-adult-child:verify-email.request-new')}</p>
+            <p className="mb-8">
+              <Trans ns={handle.i18nNamespaces} i18nKey="apply-adult-child:verify-email.unable-to-verify" components={{ communicationLink }} />
+            </p>
+            <p className="mb-4 italic">{t('apply:required-label')}</p>
+            <div className="grid items-end gap-6 md:grid-cols-2">
+              <InputField
+                id="verification-code"
+                name="verificationCode"
+                className="w-full"
+                errorMessage={errors?.verificationCode}
+                label={t('apply-adult-child:verify-email.verification-code-label')}
+                aria-describedby="verification-code"
+                inputMode="numeric"
+                required
+              />
+            </div>
+            <LoadingButton
+              id="request-button"
+              type="button"
+              name="_action"
+              variant="link"
+              loading={isSubmitting}
+              value={FORM_ACTION.request}
+              data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form-Adult-Child:Request new verification code - Verify email click"
```

Comment:
To be consistent with the rest of the adult-child flow:

```suggestion
              data-gc-analytics-customclick="ESDC-EDSC:CDCP Online Application Form-Adult_Child:Request new verification code - Verify email click"
```

(same comment for the rest of the `customclick` attribute values in this file)